codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestDecoderTest.java

```diff
@@ -486,4 +486,23 @@ private static void testInvalidHeaders0(String requestStr) {
         assertTrue(request.decoderResult().cause() instanceof IllegalArgumentException);
         assertFalse(channel.finish());
     }
+
+    @Test
+    public void testTransferCodingGZIPAndChunked() {
+        String requestStr = "POST / HTTP/1.1\r\n" +
+                "Host: example.com\r\n" +
+                "Content-Type: application/x-www-form-urlencoded\r\n" +
+                "Transfer-Encoding: gzip, chunked\r\n" +
+                "\r\n" +
+                "71\r\n";
```

Comment:
Consider writing the full HTTP request (including one or two chunks + last (zero) chunk + some trailers) to make sure that decoder's state machine transitions to the correct state and handles the entire request correctly.

Reply:
Done by move the tests to `HttpContentDecoderTest` to use the pre-defined `gzipped hello world` for testing.